api/apps/api/src/modules/projects/planning-unit-grid/set-project-grid-from-shapefile.handler.ts

```diff
@@ -48,6 +51,18 @@ export class SetProjectGridFromShapefileHandler
       await manager
         .getRepository(PlanningArea)
         .update({ id: planningAreaId }, { projectId });
+      const puIds = await manager
+        .getRepository(ProjectsPuEntity)
+        .find({ where: { projectId } })
+        .then((pu) => pu.map((p) => p.id));
+      const costSurfaceRepository = manager.getRepository(
+        CostSurfacePuDataEntity,
+      );
+      await Promise.all(
+        puIds.map((puid) =>
+          costSurfaceRepository.update({ puid }, { costSurfaceId }),
+        ),
```

Comment:
Nicely spotted.  I will try to bullet point the ideas to be very explicit.

1. We marked `cost_surface_put_data.cost_surface_id` as NOT NULL (probably a bad decision that we could rollback, because projects_pu.project_id can actually be NULL, and its left NULL until it gets updated for this case. However, for other cases project_id is set to planning_area_id until it gets updated. Yes, I know)

2. This code is triggered after a shapefile upload with a grid (no grid is another story). So before this code is run we have:
 - planning_area
 -projects_pu
 -cost_surface_pu
 
Everything we do is to link the entities among both databases, hence the `update`/

3. We select the all the PUs (project_pu.id) for the given project, and each of these PU has a cost, associated again to the given project. Since the cost is associated to a given project, and this project has at this point only one (the default) cost surface, all cost related to these PU, MUST be associated to this api.cost_surface

I hope this helps, we can of course sync later on if something is yet not clear